File: backend/src/main/kotlin/ase/athlete_view/domain/activity/pojo/entity/Lap.kt
package ase.athlete_view.domain.activity.pojo.entity

import ase.athlete_view.domain.activity.pojo.util.StepType
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
@Entity
class Lap(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long?= null,
    var lapNum: Int,
    var time: Int, // in ds (10^-1 s)
    var distance: Int, // in m
    var avgSpeed: Float, // in m/s
    var avgPower: Int, // in W
    var maxPower: Int, // in W
    var avgBpm: Int,
    var maxBpm: Int,
    var avgCadence: Int,
    var maxCadence: Int,
    var stepType: StepType
) {
    override fun toString(): String {
        return "Lap(id=$id, lapNum=$lapNum, time=$time, distance=$distance, avgSpeed=$avgSpeed, avgPower=$avgPower, maxPower=$maxPower, avgBpm=$avgBpm, maxBpm=$maxBpm, avgCadence=$avgCadence, maxCadence=$maxCadence, stepType=$stepType"
    }
}

File: frontend/src/app/common/interval/dto/Lap.ts
import { StepTargetType } from "./Step";

export interface Lap {
  id?: number;
  lapNum: number;
  time?: number; // in ds (10^-1 s)
  distance?: number; // in m
  avgSpeed?: number; // in m/s
  avgPower?: number; // in W
  maxPower?: number; // in W
  avgBpm: number;
  maxBpm: number;
  avgCadence?: number;
  maxCadence?: number;
  stepType?: StepTargetType;
}
